lib/flood_risk_engine/services/address/deserialize/ea_facade_to_address.rb

```diff
@@ -0,0 +1,81 @@
+# Specific converter - convert EA AddressFacade into other usable forms
+
+module FloodRiskEngine
+  module Services
+    module Address
+      module Deserialize
+
+        class EaFacadeToAddress < Deserialize::DeserializeBase
+
+          def selectable_address_data(inbound)
+            addresses = inbound["results"] if inbound && inbound.is_a?(Hash)
+
+            addresses ||= []
+
+            results = addresses.collect do |a|
+              {
+                moniker: a["address"],
+                uprn: a["uprn"]
+              }
+            end
+
+            results
+          end
+
+          # Builds collection of attributes in correct format to simply
+          # pass to Address new or create - to build an Address (plus Location)
+
+          def address_data(inbound) # rubocop:disable Metrics/MethodLength:
+            addresses = inbound["results"] if inbound
+
+            addresses ||= []
+
+            results = addresses.collect do |a|
+              a.default = ""
+
+              {
+                premises: a["premises"],
+                street_address: a["street_address"],
+                locality: a["locality"],
+                city: a["city"],
+                postcode: a["postcode"],
+                organisation: a["organisation"] || "",
+                state_date: a["state_date"],
+                blpu_state_code: a["blpu_state_code"],
+                postal_address_code: a["postal_address_code"],
+                logical_status_code: a["logical_status_code"],
+                uprn: a["uprn"]
+              }
+            end
+
+            results
+          end
+
+          def location_data(inbound)
+            addresses = inbound["results"] if inbound
+
+            addresses ||= []
+
+            results = addresses.collect do |a|
+              a.default = ""
+              {
+                easting: EaFacadeToAddress.format_xy(a["x"]),
+                northing: EaFacadeToAddress.format_xy(a["y"])
+              }
+            end
+
+            results
+          end
+
+          def self.format_xy(inbound)
```

Comment:
Can `format_xy` be an instance method? It would simplify the `location_data` method if it was.